pkg/controller/pipe/integration.go

```diff
@@ -210,6 +218,27 @@ func CreateIntegrationFor(ctx context.Context, c client.Client, binding *v1.Pipe
 	return &it, nil
 }
 
+// shouldHandleNoErrorHandler determines the runtime version and checks on noErrorHandler that is configured for this version.
+func shouldHandleNoErrorHandler(ctx context.Context, c client.Client, it *v1.Integration) bool {
+	var runtimeVersion string
+	if it.Spec.Traits.Camel != nil && it.Spec.Traits.Camel.RuntimeVersion != "" {
+		runtimeVersion = it.Spec.Traits.Camel.RuntimeVersion
+	} else if pl, err := platform.GetForResource(ctx, c, it); err == nil && pl != nil {
+		runtimeVersion = pl.Status.Build.RuntimeVersion
+	}
+
+	if runtimeVersion != "" {
+		runtimeVersion, _ = strings.CutSuffix(runtimeVersion, "-SNAPSHOT")
+		if versionNumber, err := strconv.Atoi(strings.ReplaceAll(runtimeVersion, ".", "")); err == nil {
```

Comment:
this would fail for those version that have suffix, ie, `3.8.0-my-companytag`.

Reply:
Should be handled by the string base comparison - will add a unit test ...